### PR TITLE
Update fuse

### PR DESCRIPTION
**Pull request by aashokvardhan**

### CODE REVIEW

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py [hunk 1]
```diff
@@ -41,6 +47,14 @@ def instantiate_system():
 
         self.assertEqual(loaded_var, new_var)
 
+    def test_delete_op(self):
+        ops = fe.op.numpyop.meta.Fuse(
+            ops=[TestNumpyOp(inputs='x', outputs='x', mode="train", var=1), Delete(
+                keys='x',
+                mode="train",
+            )])
+        _ = ops.forward(data=[self.tf_data], state={})
```
Comment (vbvg2008) [inline test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:56]: need to assert that key `x` is indeed deleted

Comment (vbvg2008) [inline test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:56]: need to assert that key `x` is indeed deleted

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py [hunk 2]
```diff
@@ -41,6 +47,14 @@ def instantiate_system():
 
         self.assertEqual(loaded_var, new_var)
 
+    def test_delete_op(self):
+        ops = fe.op.numpyop.meta.Fuse(
+            ops=[TestNumpyOp(inputs='x', outputs='x', mode="train", var=1), Delete(
```
Comment (TortoiseHam) [inline test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:52]: I think the real test we are interested in here is Fuse((a->a,b), (del b)) and make sure that it runs without requiring b as an input to Fuse

Comment (aashokvardhan) [reply test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:52]: Sure will add that test case too.

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py (hunk 1 above)
Comment (aashokvardhan) [reply test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:56]: With the changes this test case is already failing but will add that assert too.

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py (hunk 2 above)
Comment (TortoiseHam) [reply test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:52]: generally it's better if unit tests are black-box such that if someone later came and messed up the code without realizing why it had been written that way the test would catch it

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py (hunk 1 above)
Comment (TortoiseHam) [reply test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:56]: the test case is failing?

Comment (aashokvardhan) [reply test/PR_test/integration_test/op/numpyop/meta/test_fuse.py:56]: I mean without the changes the unit test would fail.

**Review by TortoiseHam — APPROVED** [review]
LG

**Pull request merged**